### PR TITLE
shotcut: 21.09.20 -> 22.12.21

### PR DESCRIPTION
**Pull request by MatthewCroughan**
###### Description of changes

Upgrades shotcut from 21.09.20 -> 22.12.21, they seem to have changed the build system from QMake to CMake. I've done a few minimal changes, which means that it builds successfully, but my derivation probably isn't quite right and requires more changes, which is why I've made this a draft PR.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/shotcut/default.nix [hunk 1]
```diff
@@ -45,14 +48,12 @@ mkDerivation rec {
     qtwebsockets
     qtquickcontrols2
     qtgraphicaleffects
+    fftw
   ];
 
   env.NIX_CFLAGS_COMPILE = "-I${mlt.dev}/include/mlt++ -I${mlt.dev}/include/mlt";
```
Comment (woffs) [inline pkgs/applications/video/shotcut/default.nix:54]: this can apparently be omitted

File: pkgs/applications/video/shotcut/default.nix [hunk 2]
```diff
@@ -45,14 +48,12 @@ mkDerivation rec {
...
-  qmakeFlags = [
-    "QMAKE_LRELEASE=${lib.getDev qttools}/bin/lrelease"
-    "SHOTCUT_VERSION=${version}"
-    "DEFINES+=SHOTCUT_NOUPGRADE"
-  ];
+
+  SHOTCUT_VERSION = "${version}";
```
Comment (woffs) [inline pkgs/applications/video/shotcut/default.nix:56]: We have to put this into cmakeFlags:
`cmakeFlags = [ "-DSHOTCUT_VERSION=${version}" ];`

**Review by woffs — CHANGES_REQUESTED** [review]
Thanks for your work! Please see my requested change so Help->About shows the correct version.

File: pkgs/applications/video/shotcut/default.nix [hunk 3]
```diff
@@ -45,14 +48,12 @@ mkDerivation rec {
...
     qtquickcontrols2
     qtgraphicaleffects
+    fftw
   ];
 
   env.NIX_CFLAGS_COMPILE = "-I${mlt.dev}/include/mlt++ -I${mlt.dev}/include/mlt";
-  qmakeFlags = [
-    "QMAKE_LRELEASE=${lib.getDev qttools}/bin/lrelease"
-    "SHOTCUT_VERSION=${version}"
-    "DEFINES+=SHOTCUT_NOUPGRADE"
```
Comment (woffs) [inline pkgs/applications/video/shotcut/default.nix:54]: We have to put this into
`env.NIX_CFLAGS_COMPILE = "-DSHOTCUT_NOUPGRADE=1";`

File: pkgs/applications/video/shotcut/default.nix (hunk 1 above)
Comment (MatthewCroughan) [reply pkgs/applications/video/shotcut/default.nix:54]: Omitting this causes the build to fail
```
error: builder for '/nix/store/15qln57gip1vlnakkiyg1pdpavvz1n7j-shotcut-22.12.21.drv' failed with exit code 1;
       last 10 log lines:
       > --   No package 'fftw' found
       > CMake Error at /nix/store/l341npqn3j8pwqq1nqgj5cqhnfbzhphd-cmake-3.25.1/share/cmake-3.25/Modules/FindPkgConfig.cmake:607 (message):
       >   A required package was not found
       > Call Stack (most recent call first):
       >   /nix/store/l341npqn3j8pwqq1nqgj5cqhnfbzhphd-cmake-3.25.1/share/cmake-3.25/Modules/FindPkgConfig.cmake:829 (_pkg_check_modules_internal)
       >   CMakeLists.txt:49 (pkg_check_modules)
       >
       > 
       > -- Configuring incomplete, errors occurred!
       > See also "/build/source/build/CMakeFiles/CMakeOutput.log".
       For full logs, run 'nix log /nix/store/15qln57gip1vlnakkiyg1pdpavvz1n7j-shotcut-22.12.21.drv'.
```

Comment (woffs) [reply pkgs/applications/video/shotcut/default.nix]: sorry, I meant the CFLAGS line and not fftw. Github point-and-click fooled me.

**Review by woffs — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Closing in favor of #269444

**Pull request closed without merging**